google-cloud-examples/src/main/java/com/google/cloud/examples/deployment-manager/DeploymentManagerDemo.java

```diff
@@ -0,0 +1,117 @@
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.deploymentmanager.DeploymentManager;
+import com.google.api.services.deploymentmanager.model.*;
+
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+
+public class DeploymentManagerDemo {
+
+    private static String PROJECTID = "project-id";
+    private static String CONFIG_YAML = "/path/to/deployment-manager-config.yaml";
+
+    // authentication uses the GOOGLE_APPLICATION_CREDENTIALS environemnt variable.
+    // run `gcloud auth application-default login` to create/update the file locally
+
+    // in intellij you can set this in the runtime config next to the run button
+    // private static String GOOGLE_APPLICATION_CREDENTIALS="/path/to/application_default_credentials.json";
+
+    // javadoc reference for deployment manger is here
+    // https://developers.google.com/resources/api-libraries/documentation/deploymentmanager/v2/java/latest/
+    public static void main(String args[]) throws GeneralSecurityException, IOException {
+
+        System.out.println("GOOGLE_APPLICATION_CREDENTIALS: " + System.getenv("GOOGLE_APPLICATION_CREDENTIALS"));
+
+
+        // best practice is to use application default credential
+        GoogleCredential credential = GoogleCredential.getApplicationDefault();
+
+        // to load from somewhere else use this
+        // GoogleCredential credential = GoogleCredential.fromStream(new FileInputStream(GOOGLE_APPLICATION_CREDENTIALS));
+
+
+        HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+
+
+        DeploymentManager dm2 = new DeploymentManager.Builder(httpTransport, jsonFactory, credential)
+                .setApplicationName("New_Application")
+                .build();
+
+        Deployment deploy = new Deployment();
+        // Optional  we want some labels: Map of labels; provided by the client when the resource is created or updated.
+        List<DeploymentLabelEntry> labels = new ArrayList<DeploymentLabelEntry>();
+
+
+        labels.add(new DeploymentLabelEntry()
+                .setKey("test-label-1")
+                .setValue("good-fella"));
+
+        deploy.setLabels(labels);
+
+
+        // load the config file contents and setup target conrfiguration
+        ConfigFile config = new ConfigFile()
+                .setContent(readYaml());
+
+
+        TargetConfiguration target = new TargetConfiguration()
+                .setConfig(config);
+
+        deploy.setTarget(target);
+        deploy.setName("my-deployment");
+
+
+        List<Deployment> deployments =
```

Comment:
Consider just doing `dm2.deployments().get(PROJECTID, "my-deployment")` and checking the return value, instead of getting all deployments then iterating over them.